docs/Containers/WireGuard.md

```diff
@@ -103,9 +103,60 @@ With most containers, you can continue to tweak environment variables and settin
+You have several options for how your remote peers resolve DNS requests:
+
+* `PEERDNS=auto`
+
+	The default value of `auto` instructs the WireGuard *service* running within the WireGuard *container* to use the same DNS as the WireGuard *container* when resolving requests from connected peers. In practice, that means the *service* directs queries to 127.0.0.11, which Docker intercepts and forwards to whichever resolvers are specified in the Raspberry Pi's `/etc/resolv.conf`.
+
+* <a name="customContInit"> `PEERDNS=auto` with `custom-cont-init` </a>
+
+	This configuration instructs WireGuard to forward DNS queries from remote peers to any **container** which is listening on port 53. This is the option you will want to choose if you are running an ad-blocking DNS server (eg *PiHole* or *AdGuardHome*) in a container on the same host as WireGuard, and you want your remote clients to obtain DNS resolution via the ad-blocker.
+
+	> Acknowledgement: thanks to @ukkopahis for developing this option.
+
+	To activate this feature:
+
+	1. Make sure your WireGuard service definition contains `PEERDNS=auto`.
+	2. Start the WireGuard container by executing:
+
+		```bash
+		$ cd ~/IOTstack
+		$ docker-compose up -d wireguard
+		```
+
+		This ensures that the `~/IOTstack/volumes/wireguard` folder structure is created and remote client configurations are (re)generated properly.
+
+	3. Run the following commands:  
 
-	If you have a local upstream DNS server, you can change this setting so that queries are directed to that server. For example:
+		```bash
+		$ cd ~/IOTstack
+		$ sudo cp ./.templates/wireguard/use-container-dns.sh ./volumes/wireguard/custom-cont-init.d/
+		$ docker-compose restart wireguard
+		```
+
+	 	The presence of `use-container-dns.sh` causes WireGuard to redirect incoming DNS queries to the default gateway on the internal bridged network. That, in turn, results in the queries being forwarded to any other container that is listening for DNS traffic on port 53. It does not matter if that other container is PiHole, AdGuardHome, bind9 or any other kind of DNS server.
+	 
+	 	Do note, however, that this configuration creates a dependency between WireGuard and the container providing DNS resolution. You may wish to make that explicit in your `docker-compose.yml` by adding these lines to your WireGuard service definition:
+	 
+		```yaml
+		depends_on:
+		  - pihole
+		```
+
+	 	> Substitute `adguardhome` or `bind9` for `pihole`, as appropriate.
+
+	Once activated, this feature will remain active until you decide to deactivate it. If you ever wish to deactivate it, run the following commands:
+
+	```bash
+	$ cd ~/IOTstack
+	$ sudo rm ./volumes/wireguard/custom-cont-init.d/use-container-dns.sh
+	$ docker-compose restart wireguard
+	```
+
+* `PEERDNS=«ip address»`
+
+	A third possibility is if you have a local upstream DNS server. You can specify the IP address of that server so that remote peers receive DNS resolution from that host. For example:
```

Comment:
But preferably let coredns do dns forwarding if there is even the slimmest of chances the DNS server IP will change. Changes to the PEERDNS are not propagated to existing clients. It's much easier to just maintain where DNS is forwarded on the server than to change every Wireguard client.